.ci/setup.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+test_repo="github.com/clearcontainers/tests"
+
+# Clone Tests repository.
+go get "$test_repo"
+
+test_repo_dir="${GOPATH}/src/${test_repo}"
+
+# Check the commits in the branch
+checkcommits_dir="${test_repo_dir}/cmd/checkcommits"
+(cd "${checkcommits_dir}" && make)
+checkcommits \
+	--need-fixes \
+	--need-sign-offs \
+	--body-length 72 \
+	--subject-length 75 \
```

Comment:
While I understand this is a general guideline for kernel commit messages, it seems a bit too strict to enforce them on every commit. What if I need to paste some logs in the commit message? It will be really strange to reformat them with new lines. For example https://github.com/kata-containers/shim/commit/33ed4a393c37b6ebe32c72787db08747c0f28587 that started the desire to include `checkcommits` in the first place.

And even Linux kernel does not enforce the body length and subject length rules. I hope we'd allow them as well.

Reply:
@bergwolf Yes, I'm fine with that. The k8s/container community is not really used to this kind of restrictions, so it would make sense to accommodate them better.

@jodh Would you mind removing those restrictions?

Reply:
@sameo - I certainly can do, but can I first check something...

@bergwolf - Have you read https://github.com/clearcontainers/tests/tree/master/cmd/checkcommits#handling-long-lines? Logs tend to start with a numeric (timestamp, pid, etc), so to deal with that `checkcommits` ignores all line lengths for lines starting with a numeric. Alternatively, you can simply add a "> " to the start of every line to force that behaviour, which would handle the precise scenario you've referred to above. Yes, there is a tiny bit more work to do, and yes, you can therefore circumvent the line length restrictions but the point of `checkcommits` is to provide a lightweight process and catch common mistakes. It's not supposed to be too rigid.

Reply:
@jodh-intel While `signed-offs` are legal and `fixes` are process, the rules on commit message length are purely to make it look pretty, which is really unnecessary and annoying. Having to workaround them is adding more difficulty IMHO. I understand that we want to follow kernel commit rules but even the Linux kernel does not have such requirements. I hope we do not make it difficult for people to contribute to the project. So please remove them.

Reply:
Sure, not a problem - branch updated ;)

Reply:
Thanks @jodh-intel !